controllers/hazelcast/wanreplication_controller.go

```diff
@@ -778,10 +781,74 @@ func getLogger(ctx context.Context) logr.Logger {
 func (r *WanReplicationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&hazelcastv1alpha1.WanReplication{}).
+		Watches(&source.Kind{Type: &hazelcastv1alpha1.Map{}}, handler.EnqueueRequestsFromMapFunc(r.beingSuccessfulMapUpdates),
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(createEvent event.CreateEvent) bool {
+					return false
+				},
+				UpdateFunc: func(updateEvent event.UpdateEvent) bool {
+					// to run handler 'beingSuccessfulMapUpdates' function after the map is ready
+					oldMap, ok := updateEvent.ObjectOld.(*hazelcastv1alpha1.Map)
+					if !ok {
+						return false
+					}
+					newMap, ok := updateEvent.ObjectNew.(*hazelcastv1alpha1.Map)
+					if !ok {
+						return false
+					}
+					return oldMap.Status.State != hazelcastv1alpha1.MapSuccess && newMap.Status.State == hazelcastv1alpha1.MapSuccess
+				},
+				DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+					return false
+				},
+				GenericFunc: func(genericEvent event.GenericEvent) bool {
+					return false
+				},
+			}),
+		).
 		Watches(&source.Kind{Type: &hazelcastv1alpha1.Map{}}, handler.EnqueueRequestsFromMapFunc(r.terminatingMapUpdates)).
 		Complete(r)
 }
 
+func (r *WanReplicationReconciler) beingSuccessfulMapUpdates(m client.Object) []reconcile.Request {
```

Comment:
```suggestion
func (r *WanReplicationReconciler) findNewlyCreatedMaps(m client.Object) []reconcile.Request {
```
or 
```suggestion
func (r *WanReplicationReconciler) findNewMaps(m client.Object) []reconcile.Request {
```
are better namings, I think

Reply:
`terminatingMapUpdates` is not also good naming; what about `findTerminatedMaps` or `findTerminatationCandidateMaps`? cc: @mtyazici

Reply:
sounds good to me 👍